src/Sentry/Internal/FileSystem.cs

```diff
@@ -16,38 +19,112 @@ public IEnumerable<string> EnumerateFiles(string path, string searchPattern) =>
     public IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption) =>
         Directory.EnumerateFiles(path, searchPattern, searchOption);
 
-    public void CreateDirectory(string path) => Directory.CreateDirectory(path);
+    public DirectoryInfo? CreateDirectory(string path)
+    {
+        if (!_options?.DisableFileWrite is false)
+        {
+            _options?.LogDebug("Skipping creating directory. Writing to file system has been explicitly disabled.");
+            return null;
+        }
```

Comment:
This works but it does make the `FileSystem` class a bit harder to read and also means we're frequently checking the value of `DisableFileWrite` (even though this never changes, once initialised). 

An alternative would be to have two different implementations of `IFileSystem`:
1. SentryReadWriteFileSystem
2. SentryReadOnlyFileSystem

`SentryOptions.DisableFileWrite` could be inspected once and used to determine which concrete type to use when initialising when initialising `SentryOptions.FileSystem`.

That's a code style choice though - up to you.